### PR TITLE
feat: digital asset deployment

### PR DESCRIPTION
**Pull request by CallumGrindle**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Feature: deployment function for LSP7/8

- **What is the current behaviour?** (You can also link to an open issue here)
- No digital asset support

- **What is the new behaviour (if this is a feature change)?**
- Add deployment function for LSP7/8. 

- **Other information**:
- There is some duplication in the way that LSP7 and LSP8 are deployed. I have kept these separate for now to avoid early abstraction as there will be additional features coming which may change how these functions work
- I added LSP7/8 classes + ABIs to `/tmp` because they have not been released on smart contracts NPM package yet, as soon as these are released we will remove these

### CODE REVIEW

File: src/lib/interfaces/digital-asset-deployment.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+export interface LSP7DigitalAssetDeploymentOptions extends DigitalAssetDeploymentOptions {
+  isNFT: boolean;
+}
+
+export enum DeploymentStatus {
+  PENDING = 'PENDING',
+  COMPLETE = 'COMPLETE',
+}
```
Comment (Hugoo) [inline src/lib/interfaces/digital-asset-deployment.ts:22]: This is duplicate with `src/lib/interfaces/deployment-events.ts:L12`

Can you use the same interface instead?

File: src/lib/classes/digital-asset.ts [hunk 2]
```diff
@@ -20,14 +22,27 @@ export class DigitalAsset {
...
-
+  deployLSP7DigitalAssetReactive(
+    digitalAssetDeploymentOptions: LSP7DigitalAssetDeploymentOptions,
+    contractDeploymentOptions?: ContractDeploymentOptions
+  ) {
+    const digitalAsset$ = lsp7DigitalAssetDeployment$(
+      this.signer,
+      digitalAssetDeploymentOptions,
+      contractDeploymentOptions?.libAddress ??
+        versions[this.options.chainId]?.baseContracts?.LSP7DigitalAsset['0.0.1']
```
Comment (Hugoo) [inline src/lib/classes/digital-asset.ts:35]: `'0.0.1'` is the contract version number?
Would it make sense to put this at the top of the file like CONTRACT_VERSION = '0.0.1' or in a const/globals file ?

File: src/lib/classes/digital-asset.ts [hunk 3]
```diff
@@ -58,20 +60,29 @@ export class DigitalAsset {
...
-    digitalAssetDeploymentOptions: DigitalAssetDeploymentOptions
+    digitalAssetDeploymentOptions: DigitalAssetDeploymentOptions,
+    contractDeploymentOptions?: ContractDeploymentOptions
   ) {
     const digitalAsset$ = lsp8IdentifiableDigitalAssetDeployment$(
       this.signer,
-      digitalAssetDeploymentOptions
+      digitalAssetDeploymentOptions,
+      contractDeploymentOptions?.libAddress ??
+        versions[this.options.chainId]?.baseContracts?.LSP8IdentifiableDigitalAsset['0.0.1']
```
Comment (Hugoo) [inline src/lib/classes/digital-asset.ts:73]: same here can this version number go in a global const variable maybe?

**Review by Hugoo — APPROVED** [review]
looks cool !

I'm thinking we should add Codecov + add a few tests in `lsp-factory.spec.ts`

File: src/lib/interfaces/digital-asset-deployment.ts (hunk 1 above)
Comment (CallumGrindle) [reply src/lib/interfaces/digital-asset-deployment.ts]: Ah cool yes i missed this, thanks :)

Comment (CallumGrindle) [reply src/lib/interfaces/digital-asset-deployment.ts]: Ah cool yes i missed this, thanks :)

**Pull request merged**